controller/contour/namespace.go

```diff
@@ -35,19 +35,16 @@ const (
 // ensureNamespace ensures the namespace for the provided name exists.
 func (r *Reconciler) ensureNamespace(ctx context.Context, name string) error {
 	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns)
-	switch {
-	case err == nil:
-		r.Log.Info("namespace exists; skipped adding", "name", ns.Name)
-		return nil
-	case errors.IsNotFound(err):
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns); err != nil {
```

Comment:
Is there still a remaining case where there's an error getting the namespace, but it's different than the `IsNotFound(...)` method case?

Reply:
@stevesloka I reorganized the conditions for both ensure namespace functions, PTAL.